daemon/k8s_watcher.go

```diff
@@ -168,18 +158,48 @@ func (d *Daemon) EnableK8sWatcher(reSyncPeriod time.Duration) error {
 	)
 	go ingressController.Run(wait.NeverStop)
 
-	_, ciliumRulesController := cache.NewInformer(
-		cache.NewListWatchFromClient(tprClient, "ciliumnetworkpolicies",
-			v1.NamespaceAll, fields.Everything()),
-		&k8sTypes.CiliumNetworkPolicy{},
-		reSyncPeriod,
-		cache.ResourceEventHandlerFuncs{
-			AddFunc:    d.addCiliumNetworkPolicy,
-			UpdateFunc: d.updateCiliumNetworkPolicy,
-			DeleteFunc: d.deleteCiliumNetworkPolicy,
-		},
-	)
-	go ciliumRulesController.Run(wait.NeverStop)
+	go func() {
```

Comment:
Could this whole function be named? Seems like quite a bit of logic (rather straightforward, but still), would be nice to have it as a separate unit (probably without `ciliumRulesController.Run(wait.NeverStop)`)